avm/src/vm/event.rs

```diff
@@ -190,7 +190,8 @@ impl HandlerFragment {
     let mut instructions = self.get_instruction_fragment();
     loop {
       // io-bound fragment
-      if !instructions[0].opcode.pred_exec {
+      match &instructions[0].opcode.fun {
```

Comment:
I've figured out why it's not working right. This `match` function matches the first instruction to determine if the fragment is an `io` or `cpu` collection, but then it uses the `async_func` or `func` from the *first instruction* on *every instruction call*, instead of using the correct function for each instruction call.